layer/filestore.go

```diff
@@ -191,6 +201,23 @@ func (fms *fileMetadataStore) GetCacheID(layer ChainID) (string, error) {
 	return content, nil
 }
 
+func (fms *fileMetadataStore) GetForeignSource(layer ChainID) (*distribution.Descriptor, error) {
+	content, err := ioutil.ReadFile(fms.getLayerFilename(layer, "foreign-source"))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	var ref *distribution.Descriptor
+	err = json.Unmarshal(content, &ref)
+	if err != nil {
+		return nil, err
+	}
+	return ref, err
```

Comment:
We return nil to mean that there is no foreign source. I think this is the right way to handle it, rather than requiring callers to check for a non-empty digest or return an additional bool or something.

You're right, though; if `foreign-source` exists, we should return a non-nil value.